rtree.go

```diff
@@ -88,24 +88,18 @@ func (s *dimSorter) Less(i, j int) bool {
 	return s.objs[i].bb.p[s.dim] < s.objs[j].bb.p[s.dim]
 }
 
-// split splits objects into slices of maximum k elements.
-func split(k int, objs []entry) [][]entry {
-	n := len(objs) / k
-	if len(objs)%k != 0 {
-		n++
-	}
+// walkPartitions splits objs into slices of maximum k elements and
+// iterates over these partitions.
+func walkPartitions(k int, objs []entry, iter func(parts []entry)) {
+	n := (len(objs) + k - 1) / k
 
-	splits := make([][]entry, n)
-	for i := 0; i < n; i++ {
-		if i == n-1 {
-			splits[i] = objs[i*k:]
-			break
-		}
-
-		splits[i] = objs[i*k : (i+1)*k]
+	for i := 1; i < n; i++ {
+		iter(objs[(i-1)*k : i*k])
 	}
 
-	return splits
+	if objs = objs[(n-1)*k:]; len(objs) != 0 {
```

Comment:
Will `objs` ever have lenght 0? You loop n-1 times, so there is always elements in this last iteration.

Reply:
Basically `n` should never be 0 here. If so, your tree parameter `M` is misconfigured, since `walkPartitions` is called only when `len(objs) > m`, where m is either 1 or `M`.

Reply:
Originally it was meant to avoid populating zero-length partitions. But you're right, this can not happen and the check is unnecessary. I'll remove it.